From b21ea85641941ae768014ca06bc61d1fbcea9a70 Mon Sep 17 00:00:00 2001
From: Ricardo Neri <ricardo.neri-calderon@linux.intel.com>
Date: Fri, 17 Jul 2015 17:11:23 -0700
Subject: [PATCH 2/4] efi: allow efi_mem_desc_lookup find
 EFI_BOOT_SERVICES_CODE regions

CONFIG_EFI_BOOT_SERVICES_WARN detects illegal accesses to EFI_BOOT_-
SERVICES_CODE/DATA memory regions using the offending physical address
of the access. We need the memory descriptor of such region so that
it can be mapped as a fixup to a page fault. These illegal accesses
can occur in both code and data regions. Thus, allow efi_mem_desc_-
lookup to return the descriptor of EFI_BOOT_SERVICES_CODE.

Signed-off-by: Ricardo Neri <ricardo.neri-calderon@linux.intel.com>
---
 drivers/firmware/efi/efi.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/firmware/efi/efi.c b/drivers/firmware/efi/efi.c
index 027ca21..0036cba 100644
--- a/drivers/firmware/efi/efi.c
+++ b/drivers/firmware/efi/efi.c
@@ -295,6 +295,9 @@ int __init efi_mem_desc_lookup(u64 phys_addr, efi_memory_desc_t *out_md)
 
 		if (!(md->attribute & EFI_MEMORY_RUNTIME) &&
 		    md->type != EFI_BOOT_SERVICES_DATA &&
+#if defined(CONFIG_EFI_BOOT_SERVICES_WARN)
+		    md->type != EFI_BOOT_SERVICES_CODE &&
+#endif
 		    md->type != EFI_RUNTIME_SERVICES_DATA) {
 			early_memunmap(md, sizeof (*md));
 			continue;
-- 
1.9.1

